financial-templates-lib/logger/SpyTransport.js

```diff
@@ -0,0 +1,26 @@
+// This transport enables unit tests to validate values passed to the winston using a Sinon Spy.
+
+const Transport = require("winston-transport");
+
+class SpyTransport extends Transport {
+  constructor(winstonOptions, spyOptions) {
+    super(winstonOptions);
+    this.spy = spyOptions.spy; // local instance of the spy to capture passed messages.
+  }
+
+  async log(info, callback) {
+    // add the info sent to the winston transport to the spy. this enables unit tests to validate what is passed to winston
+    this.spy(info);
+    callback();
+  }
+}
+
+// Helper function used by unit tests to check if the last message sent to winston contains a particular string value.
+// Caller feeds in the spy instance and the value to check.
+const lastSpyLogIncludes = (spy, value) => {
+  // Sinon's getCall(n) function returns the values sent in in the nth call the the spy
+  const lastReturnedArg = spy.getCall(-1).lastArg.mrkdwn.toString();
+  return lastReturnedArg.indexOf(value) != -1;
```

Comment:
Just to check, `indexOf(value)` will return a positive index if `value` is in the `mrkdwn` string?

Reply:
yup